plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/ArchiveCaseAction.java

```diff
@@ -0,0 +1,247 @@
+package ca.on.oicr.gsi.shesmu.nabu;
+
+import ca.on.oicr.gsi.prometheus.LatencyHistogram;
+import ca.on.oicr.gsi.shesmu.plugin.*;
+import ca.on.oicr.gsi.shesmu.plugin.action.*;
+import ca.on.oicr.gsi.shesmu.plugin.json.JsonListBodyHandler;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.prometheus.client.Counter;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class ArchiveCaseAction extends JsonParameterisedAction {
+
+  private final Definer<NabuPlugin> owner;
+  static final ObjectMapper MAPPER = new ObjectMapper();
+  private List<String> errors = List.of();
+  public String caseId;
+  public long requisitionId;
+  public Set<String> limsIds;
+  public Set<String> workflowRunIdsForOffsiteArchive;
+  public Set<String> workflowRunIdsForVidarrArchival;
+  static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+
+  private static final Counter NabuRequestErrors =
+      Counter.build(
+              "shesmu_nabu_request_errors",
+              "The number of errors trying to countact the Nabu web service.")
+          .labelNames("target")
+          .register();
+  private static final LatencyHistogram NabuRequestTime =
+      new LatencyHistogram(
+          "shesmu_nabu_request_time",
+          "The request time latency to launch a remote action.",
+          "target");
+
+  public ArchiveCaseAction(Definer<NabuPlugin> owner) {
+    super("nabu-plugin");
+    this.owner = owner;
+  }
+
+  @ActionParameter
+  public void caseId(String caseId) {
+    this.caseId = caseId;
+  }
+
+  @ActionParameter
+  public void requisitionId(long requisitionId) {
+    this.requisitionId = requisitionId;
+  }
+
+  @ActionParameter
+  public void workflowRunIdsForOffsiteArchive(Set<String> workflowRunIdsForOffsiteArchive) {
+    this.workflowRunIdsForOffsiteArchive = workflowRunIdsForOffsiteArchive;
+  }
+
+  @ActionParameter
+  public void workflowRunIdsForVidarrArchival(Set<String> workflowRunIdsForVidarrArchival) {
+    this.workflowRunIdsForVidarrArchival = workflowRunIdsForVidarrArchival;
+  }
+
+  @ActionParameter
+  public void limsIds(Set<String> limsIds) {
+    this.limsIds = limsIds;
+  }
+
+  @Override
+  public ObjectNode parameters() {
+    return null;
```

Comment:
From the abstract class you're extending: https://github.com/oicr-gsi/shesmu/blob/master/shesmu-pluginapi/src/main/java/ca/on/oicr/gsi/shesmu/plugin/action/JsonParameterisedAction.java#L16C1-L16C35
"This must not return null."

Check out the other two children of JsonParameterisedAction for how to implement this